keras/applications/mobilenet.py

```diff
@@ -219,10 +219,15 @@ def MobileNet(input_shape=None,
                              '`0.25`, `0.50`, `0.75` or `1.0` only.')
 
         if rows != cols or rows not in [128, 160, 192, 224]:
-            raise ValueError('If imagenet weights are being loaded, '
-                             'input must have a static square shape (one of '
-                             '(128,128), (160,160), (192,192), or (224, 224)).'
-                             ' Input shape provided = %s' % (input_shape,))
+            if rows is None:
+                rows = 224
+                warnings.warn('The MobileNet shape is undefined.'
+                              ' Weights with input shape (224, 224) will be loaded.')
+            else:
+                raise ValueError('If imagenet weights are being loaded, '
+                                 'input must have a static square shape (one of '
+                                 '(128,128), (160,160), (192,192), or (224, 224)).'
```

Comment:
Format shape tuples with one space after the comma: `(128, 128)`, etc. (this style issue was pre-existing in the original code).